docs/how-to/troubleshoot_deployment_issues.md

```diff
@@ -0,0 +1,199 @@
+# Troubleshoot the deployment issues
+
+This guide provides step-by-step troubleshooting actions to remediate deployment issues. We hope you don't need this guide. If you encounter an issue and aren't able to address it via this guide, please raise an issue [here][Bug Report].
+
+## 1. Terraform failed to deploy the Charmed Aether SD-Core because of configuration mismatch
+
+### Symptoms
+
+Terraform may fail to deploy the Charmed Aether SD-Core modules. When the `terraform apply -auto-approve` command is run, the deployment fails with the wrong integration endpoints or the wrong object attributes etc. as following:
+
+```console
+$ terraform apply -auto-approve 
+╷
+│ Error: Unsupported attribute
+│ 
+│   on main.tf line 54, in resource "juju_integration" "cu-nms":
+│   54:     endpoint = module.sdcore.fiveg_identity_endpoint
+│     ├────────────────
+│     │ module.sdcore is a object
+│ 
+│ This object does not have an attribute named "fiveg_identity_endpoint".
+```
+
+### Recommended Actions
+
+[Validate the Terraform modules][Validate Terraform Configuration]:
+
+```shell
+$ terraform validate
+Success! The configuration is valid.
+```
+
+The command should not fail. If it does, please open a [bug report][Bug Report].
+
+## 2. Charmed Aether SD-Core charms stuck at the Waiting/Error status
+
+### Symptoms
+
+After deploying the Charmed Aether SD-Core, check the applications' status:
+
+```shell
+$ juju status
+```
+
+In the command output, all the charms except the `grana-agent` should be in the Active/Idle state. If any application's status is `Waiting` or `Error`, please apply the recommended actions.
+
+### Recommended Actions
+
+if any application hangs in `Waiting` or `Error` status, please open a [bug report][Bug Report].
+
+| Charm Status | How much time passed                             | Reason                | Recommended Action                                                    |
+|--------------|--------------------------------------------------|-----------------------|-----------------------------------------------------------------------|
+| Waiting      | Time is exceeded according the followed tutorial |                       | File a bug report                                                     | 
+| Error        |                                                  | Any reason            | File a bug report                                                     |
+
+Get the application container's logs:
+
+```shell
+$ microk8s.kubectl logs -f <pod_name> -c <container_name> -c <model_name>
```

Comment:
```suggestion
$ microk8s.kubectl logs -f <pod_name> -n <model_name>
```

Reply:
I think it would be better to skip the container part. Some Pods (like UPF) have multiple containers and the user may not know which one is relevant.

Reply:
OK.

Reply:
Done